internal/services/users/user_resource_msgraph.go

```diff
@@ -0,0 +1,285 @@
+package users
+
+import (
+	"context"
+	"errors"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/manicminer/hamilton/msgraph"
+
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	helpers "github.com/hashicorp/terraform-provider-azuread/internal/helpers/msgraph"
+	"github.com/hashicorp/terraform-provider-azuread/internal/tf"
+	"github.com/hashicorp/terraform-provider-azuread/internal/utils"
+)
+
+func userResourceCreateMsGraph(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client).Users.MsClient
+
+	upn := d.Get("user_principal_name").(string)
+	mailNickName := d.Get("mail_nickname").(string)
+
+	// default mail nickname to the first part of the UPN (matches the portal)
+	if mailNickName == "" {
+		mailNickName = strings.Split(upn, "@")[0]
```

Comment:
whilst there should be one @ here, is there a crash here?

Reply:
AFAICT, if the "@" is absent, or even for empty strings, indexing the zeroth item always seems to succeed